packages/x-scheduler/src/primitives/utils/TimeoutManager.ts

```diff
@@ -0,0 +1,49 @@
+export class TimeoutManager {
```

Comment:
Since all our logic is centralized in a single big store class, I think it's worth having a clean timeout and interval manager from the start, to make sure we clean all of them when unmounting the component.

This class is adapter from Romain's `useTimeout` hook to handle several timeouts and handle intervals.